File: VB/RestrictedDraggingExample/MainForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.XtraEditors
Imports System.Collections.Generic
Imports DevExpress.XtraBars.Docking
Imports RestrictedDraggingExample.Docking

Namespace RestrictedDraggingExample
	Partial Public Class MainForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			DockingManager.ForceInitialize()
			Dim dp As DockPanel = New CustomDockPanel(DockingManager)
			dp.DockTo(DockingStyle.Float)
			dp.Text = "Test Panel"
			dp.Parent.Location = PointToScreen(Point.Empty)
		End Sub
	End Class
End Namespace

File: CS/RestrictedDraggingExample/MainForm.cs
using System;
using System.Drawing;
using DevExpress.XtraEditors;
using System.Collections.Generic;
using DevExpress.XtraBars.Docking;
using RestrictedDraggingExample.Docking;

namespace RestrictedDraggingExample {
    public partial class MainForm :XtraForm {
        public MainForm() {
            InitializeComponent();
            DockingManager.ForceInitialize();
            DockPanel dp = new CustomDockPanel(DockingManager);
            dp.DockTo(DockingStyle.Float);
            dp.Text = "Test Panel";
            dp.Parent.Location = PointToScreen(Point.Empty);
        }
    }
}